Source/VMware.vSphereDSC/DSCResources/PowerCLISettings.ps1

```diff
@@ -0,0 +1,235 @@
+        }
+
+        if ($this.DefaultVIServerMode -ne [DefaultVIServerMode]::Unset) {
+            $powerCLIConfigurationProperties.Add("DefaultVIServerMode", $this.DefaultVIServerMode)
+        }
+
+        if ($null -ne $this.DisplayDeprecationWarnings) {
+            $powerCLIConfigurationProperties.Add("DisplayDeprecationWarnings", $this.DisplayDeprecationWarnings)
+        }
+
+        if ($this.InvalidCertificateAction -ne [BadCertificateAction]::Unset) {
+            $powerCLIConfigurationProperties.Add("InvalidCertificateAction", $this.InvalidCertificateAction)
+        }
+
+        if ($null -ne $this.ParticipateInCeip) {
+            $powerCLIConfigurationProperties.Add("ParticipateInCeip", $this.ParticipateInCeip)
+        }
+
+        if ($this.ProxyPolicy -ne [ProxyPolicy]::Unset) {
+            $powerCLIConfigurationProperties.Add("ProxyPolicy", $this.ProxyPolicy)
+        }
+
+        if ($null -ne $this.WebOperationTimeoutSeconds) {
+            $powerCLIConfigurationProperties.Add("WebOperationTimeoutSeconds", $this.WebOperationTimeoutSeconds)
+        }
+
+        return $powerCLIConfigurationProperties
+    }
+
+    <#
+    .DESCRIPTION
+
+    Constructs the Set-PowerCLIConfiguration cmdlet with the passed properties.
+    #>
+    [string] ConstructCommandWithParameters($commandName, $properties, $namesOfProperties) {
+        $constructedCommand = ''
+
+        # Adds the command name to the constructed command.
+        $constructedCommand += "$commandName "
+
+        # For every property name we add the property value with the following syntax: '-Property Value'.
+        foreach ($propertyName in $namesOfProperties) {
+            $propertyValue = $properties.$propertyName
+
+            # For bool values we need to add another '$' sign so the value can be evaluated to bool.
+            if ($propertyValue.GetType().Name -eq 'Boolean') {
+                $constructedCommand += "-$propertyName $"
+                $constructedCommand += "$propertyValue "
+            }
+            else {
+                $constructedCommand += "-$propertyName $propertyValue "
+            }
+        }
+
+        # Add the confirm:$false to the command to ignore the confirmation.
+        $constructedCommand += "-Confirm:$"
+        $constructedCommand += "$false "
+
+        # Trim trailing whitespaces at the end of the command.
+        $constructedCommand = $constructedCommand.TrimEnd()
+
+        return $constructedCommand
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns a boolean value indicating if the current PowerCLI Configuration is equal to the Desired Configuration.
+    #>
+    [bool] Equals($powerCLICurrentConfiguration, $powerCLIDesiredConfiguration) {
+        foreach ($key in $powerCLIDesiredConfiguration.Keys) {
+            if ($powerCLIDesiredConfiguration.$key.ToString() -ne $powerCLICurrentConfiguration.$key.ToString()) {
```

Comment:
This one is tricky as well, imagine hashtable.tostring(), it's always 'System.Collections.Hashtable' while the entries could be different. It will work for this private case of settings, it could fail in the future.

Reply:
I will document in the code for which types the logic works at the moment.